haddock-library/src/Documentation/Haddock/Parser.hs

```diff
@@ -422,6 +423,55 @@ takeIndent = do
   indent <- takeHorizontalSpace
   "\n" *> takeIndent <|> return indent
 
+-- | Provides support for simple tables.
+--
+-- Tables are composed by an optional header and body. The header is composed by
+-- a single row. The body is composed by a non-empty list of rows.
+--
+-- Example table with header:
+--
+-- > +----------+----------+
+-- > | /32bit/  |   64bit  |
+-- > +==========+==========+
+-- > |  0x0000  | @0x0000@ |
+-- > +----------+----------+
+table :: Parser (DocH mod Identifier)
+table = do
+  parseTableRowDivider
+  mHeader <- optional parseTableHeader
+  content <- parseTableContent
+  return $ DocTable (Table mHeader content)
+
+parseTableHeader :: Parser [DocH mod Identifier]
+parseTableHeader = parseTableRow <* parseTableHeaderDivider
+
+parseTableContent :: Parser [[DocH mod Identifier]]
+parseTableContent = many1 (parseTableRow <* parseTableRowDivider)
+
+parseTableRow :: Parser [DocH mod Identifier]
```

Comment:
I think you should provide some way to escape `|` in table cells. Otherwise `|` results in odd parser errors.